src-v2/Judge/NTS.Judge.Blazor/Pages/Dashboard/Components/Actions/EliminationsPanel.razor

```diff
@@ -1,133 +1,68 @@
+    }
+    private string? _value;
     private MudToggleGroup<string> toggleGroup = default!;
     private Participation _selectedParticipation = default!;
-    private bool _reasonActive = false;
-    private string? _reason = default!;
-    private bool _selectActive = false;
-    private string? _select = default!;
-    private bool _toggleFormActive { get; set; } = false;
-    private bool _participationRevoked = false;
-    private QualificationRevokeType? _qualificationRevokeType = null; 
     private string SmallTextClass = "font-size-10";
-    private string ToggleGroupSelection = "";
 
-    protected void BeforeRender()
+    protected override async Task OnInitializedAsync()
     {
-        _selectedParticipation = _participationBehind.SelectedParticipation!;
-        _participationRevoked = _selectedParticipation.IsNotQualified;
-        if (_participationRevoked)
-        {
-            _toggleFormActive = true;
-            var notQualifiedCode = _selectedParticipation.NotQualified!.ToString();
-            var codes = notQualifiedCode.Split(" ");
-            _select =  codes.Length > 1 ? codes[1] : null ;
-            _reason = _selectedParticipation.NotQualified.Complement;
-            if (_reason != null)
-            {
-                _reasonActive = true;
-            }
-            else
-            {
-                _reasonActive = false;
-            }
-            if (_select != null)
-            {
-                _selectActive = true;
-            }
-            else
-            {
-                _selectActive = false;
-            }
-            ToggleButtonProgrammatically(codes, _reason);
-        }
+        await Observe(_participationBehind);
     }
 
     public void OnToggle(string value)
     {
-        _reasonActive = false;
-        _selectActive = false;
-        ToggleGroupSelection = value;
-        if (value != null)
-        {
-            _toggleFormActive = true;
-            _qualificationRevokeType = (QualificationRevokeType)Enum.Parse(typeof(QualificationRevokeType), value);
-        }
-        else
-        {
-            _toggleFormActive = false;
-        }
-        if (value == "FinishNotRanked" || value == "Disqualify")
-        {
-            _reasonActive = true;
-        }
-        else if (value == "FaiToQualify")
-        {
-            _selectActive = true;
-        }
-    }
-
-    public async void Submit(FormData data)
-    {
-        if (data.Codes.Contains(FTQCodes.FTC))
-        {
-            _qualificationRevokeType = QualificationRevokeType.FailToCompleteLoop;
-        }
-        if (_qualificationRevokeType == null)
-        {
-            await _participationBehind.RestoreQualification(data.TandemNumber);
-        }
-        else
-        {
-            await _participationBehind.RevokeQualification(data.TandemNumber, (QualificationRevokeType)_qualificationRevokeType, data.Reason, data.Codes.ToArray());
-        }
+        _buttonValue = value;
```

Comment:
When this is done, a bug appears that causes the application to freeze when Eliminate button is clicked. When reverting to separate Value and ValueChanged the bug doesn't occur.

Reply:
I don't think that's what was causing the bug. 

An interesting fact is that if you try to use `@bind-Value` while having `ValueChanged` set as well you get a compiler error that `ValueChanged` was used more than once. That is because `@bind-Value` is essentially a short-hand syntax for what you have made yourself:
1. Sets `Value` to the property
2. Creates an EventCallback that updates the value of the property
3. Sets `ValueChanged` to the previously created event callback. 

That is why you need to use a property with public getter and setter in order to use `@bind-Value`.